src/jetstream_types.rs

```diff
@@ -553,23 +584,214 @@ pub struct NextRequest {
 }
 
 #[derive(Debug, Default, Serialize, Deserialize, Clone)]
-pub(crate) struct StreamRequest {
+pub(crate) struct StreamNamesRequest {
     #[serde(default, skip_serializing_if = "is_default")]
     pub subject: String,
 }
 
-/// options for subscription
 #[derive(Debug, Default, Serialize, Deserialize, Clone)]
-pub(crate) struct SubOpts {
-    // For attaching.
-    pub stream: String,
-    pub consumer: String,
-    // For pull based consumers, batch size for pull
-    pub pull: usize,
-    // For manual ack
-    pub mack: bool,
-    // For creating or updating.
-    pub cfg: ConsumerConfig,
+pub(crate) struct StreamNamesResponse {
+    pub streams: Vec<String>,
+}
+
+/// Options for subscription
+#[derive(Debug, Default, Serialize, Deserialize, Clone)]
+pub struct SubscribeOptions {
+    // For consumer binding:
+    pub(crate) bind_only: bool,
+    pub(crate) stream_name: Option<String>,
+    pub(crate) consumer_name: Option<String>,
+
+    // For consumer configuration:
```

Comment:
Because defaults don't always indicate intent, configuration is duplicated here as optional values.